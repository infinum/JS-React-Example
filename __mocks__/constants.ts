export const TEST_INVALID_EMAIL = 'Email User not found';

export const TEST_INVALID_PASSWORD = 'Password Invalid password';

export const TEST_FLIGHT_NAME = 'Password Invalid password';
