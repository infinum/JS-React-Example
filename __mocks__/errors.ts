export const invalidEmailMessage = 'Email User not found';

export const invalidPasswordMessage = 'Password Invalid password';
