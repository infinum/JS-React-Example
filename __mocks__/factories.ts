import { createClient } from '@/datx/create-client';
import { Flight } from '@/models/Flight';
import { Session } from '@/models/Session';
import { User } from '@/models/User';
import { createFactory, oneOf, perBuild, sequence } from '@datx/test-data-factory';

const client = createClient();
const factory = createFactory(client);

export const flightFactory = factory(Flight, {
	fields: {
		id: sequence(),
		name: 'Air Force One',
		arrivesAt: '2021-01-01T00:00:00.000Z',
		airplaneModel: 'Boeing 747',
		basePrice: '100',
		currentSeatPrice: '100',
		departsAt: '2021-01-01T00:00:00.000Z',
	},
});

export const userFactory = factory(User, {
	fields: {
		id: sequence(),
		firstName: oneOf('Johnny', 'Michael', 'Philip', 'Darius', 'Nieves', 'Christian'),
		lastName: oneOf('Smith', 'Brown', 'Miller'),
	},
});

export const sessionFactory = factory(Session, {
	fields: {
		id: sequence(),
		email: sequence((x) => `user${x}@example.com`),
		user: perBuild(() => userFactory()),
	},
});
