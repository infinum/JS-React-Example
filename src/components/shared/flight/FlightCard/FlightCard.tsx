import React, { FC } from 'react';
import { Heading, Stat, StatNumber, StatHelpText, Flex } from '@chakra-ui/react';
import { format } from 'date-fns';

import { Flight } from '@/resources/Flight';

interface IFlightCardProps {
	flight: Flight;
}

export const FlightCard: FC<IFlightCardProps> = ({ flight, ...rest }) => (
	<Flex align="center" p={5} borderWidth="1px" shadow="md" {...rest}>
		<Heading as="h2" w="50%" mb={2} size="lg">
			{flight.name}
		</Heading>
		<Stat w="50%">
			<StatNumber>${flight.currentSeatPrice}</StatNumber>
			<StatHelpText>
				{format(new Date(flight.departsAt), 'MMM dd')} - {format(new Date(flight.arrivesAt), 'MMM dd')}
			</StatHelpText>
		</Stat>
	</Flex>
);
