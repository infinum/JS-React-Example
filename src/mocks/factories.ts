import { createClient } from '@/datx/create-client';
import { Session } from '@/models/Session';
import { User } from '@/models/User';
import { createFactory, oneOf, perBuild, sequence } from '@datx/test-data-factory';

const client = createClient();
const factory = createFactory(client);

export const userFactory = factory(User, {
	fields: {
		id: sequence(),
		firstName: oneOf('Johnny', 'Michael', 'Philip', 'Darius', 'Nieves', 'Christian'),
		lastName: oneOf('Smith', 'Brown', 'Miller'),
	},
});

export const sessionFactory = factory(Session, {
	fields: {
		id: sequence(),
		email: sequence((x) => `user${x}@example.com`),
		user: perBuild(() => userFactory()),
	},
});
