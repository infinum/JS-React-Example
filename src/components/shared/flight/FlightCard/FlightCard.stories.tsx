import { ComponentMeta, ComponentStory } from '@storybook/react';

import { flightFactory } from '__mocks__/factories';
import { FlightCard } from './FlightCard';

export default {
	title: 'UI/shared/flight/FlightCard',
	component: FlightCard,
} as ComponentMeta<typeof FlightCard>;

const Template: ComponentStory<typeof FlightCard> = (args) => <FlightCard {...args} />;

export const Basic = Template.bind({});
Basic.args = {
	flight: flightFactory(),
};
