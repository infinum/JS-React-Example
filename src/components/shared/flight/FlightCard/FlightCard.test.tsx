import { render, screen } from 'test-utils';
import { flightFactory } from '__mocks__/factories';
import { FlightCard } from '@/components/shared/flight/FlightCard/FlightCard';

describe('FlightCard', () => {
	beforeEach(() => {
		flightFactory.reset();
	});

	it('should render flights', () => {
		const data = flightFactory();

		render(<FlightCard flight={data} />);

		expect(screen.getByText(data.name)).toBeInTheDocument();
		expect(screen.getByText(`$${data.currentSeatPrice}`)).toBeInTheDocument();
	});
});
