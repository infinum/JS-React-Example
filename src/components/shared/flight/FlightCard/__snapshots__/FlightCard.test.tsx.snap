// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCardFallback should render fallback 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    border-color: var(--skeleton-start-color);
    background: var(--skeleton-start-color);
  }

  to {
    border-color: var(--skeleton-end-color);
    background: var(--skeleton-end-color);
  }
}

.emotion-0 {
  --skeleton-start-color: var(--chakra-colors-gray-100);
  --skeleton-end-color: var(--chakra-colors-gray-400);
  background: var(--skeleton-start-color);
  border-color: var(--skeleton-end-color);
  opacity: 0.7;
  border-radius: var(--chakra-radii-sm);
  -webkit-animation: 0.8s linear infinite alternate animation-0;
  animation: 0.8s linear infinite alternate animation-0;
  box-shadow: var(--chakra-shadows-none);
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  cursor: default;
  color: var(--chakra-colors-transparent);
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: var(--chakra-sizes-full);
  height: 100px;
  padding: var(--chakra-space-5);
}

.chakra-ui-dark .emotion-0:not([data-theme]),
[data-theme=dark] .emotion-0:not([data-theme]),
.emotion-0[data-theme=dark] {
  --skeleton-start-color: var(--chakra-colors-gray-800);
  --skeleton-end-color: var(--chakra-colors-gray-600);
}

.emotion-0::before,
.emotion-0::after,
.emotion-0 * {
  visibility: hidden;
}

<div
    class="chakra-skeleton emotion-0"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
